data/caasp/update.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+set -ex
+
+# 2.0 -> 2.0 http://download.suse.de/ibs/Devel:/CASP:/2.0:/ControllerNode:/TestUpdate/standard/Devel:CASP:2.0:ControllerNode:TestUpdate.repo
+# 
+
+while getopts "r:" opt; do
+case $opt in
+	r)
+		REPO=$OPTARG
+		;;
+	:)
+		echo "Option -$opt requires an argument." >&2
+		;;
+esac
+done
+
+saltid=$(docker ps | grep salt-master | awk '{print $1}')
+where="-P roles:(admin|kube-(master|minion))"
+srun="docker exec -i $saltid salt --batch 7"
+runner="$srun $where cmd.run"
+
+# Remove non-existent ISO repository
+# $runner "zypper rr 1"
+
+# Add repository as system venodors
```

Comment:
For QAM, we are using (in the settings) a variable called `INCIDENT_REPO` which includes the *testing repository* for our maintenance updates. Could you please take this into account? 

For example, you can *add* an *extra check*: 

if the variable `INCIDENT_REPO` exists, then add its value as a repository to all the nodes.
 like . 

A typical value for this variable would be sth like `http://download.suse.de/ibs/SUSE:/Maintenance:/6046/SUSE_Updates_SLE-SERVER_12-SP3_x86_64/` so I guess we need to put also a name for this repository. Let's name it `QAM_UPDATE`. So I guess, the command would be: 

```
my $repo = get_required_var('INCIDENT_REPO');
$runner "zypper ar --refresh --no-gpgcheck $repo QAM_UPDATE"
```

Reply:
let's discuss this on call tomorrow, I have no idea about how QAM tests are scheduled in openQA

Reply:
There are actually `MAINT_TEST_REPO` and `INCIDENT_REPO` variables (with `INCIDENT_PATCH` holding repo name) that can list repositories to use for testing.

Something might have changed, too, because I cannot find the former being used, instead there are `*_TEST_ISSUES` and `*_TEST_TEMPLATE` variables, e.g.
* `OS_TEST_ISSUES` holding repo IDs `1234,1111,0000,1212,1984,2580,8068`
* `OS_TEST_TEMPLATE` with template, e.g. `http://example.tld/ibs/SUSE:/Maintenance:/%INCIDENTNR%/SUSE_Updates_SLE-SERVER_12-LTSS_x86_64/`

`%INCIDENTNR%` in `_TEMPLATE` would be replaced by IDs in `_ISSUES`

Example code for `MAINT_TEST_REPO` to show how multiple repositories can be handled:
```perl
my @repos = split(/,/, get_var('MAINT_TEST_REPO'));
foreach my $repo (@repos) {
     assert_script_run $salt . " cmd.run \"zypper ar -G $repo $repo\"";
}
```
I think `INCIDENT_REPO` will ever hold only one repo, so that would be used for single incidents.

Reply:
Ok, I used INCIDENT_REPO variable (it's enough to set on controller node)